src/main/java/com/google/cloud/spanner/pgadapter/statements/IntermediateStatement.java

```diff
@@ -165,13 +207,21 @@ public Exception getException() {
    *
    * @throws SQLException If an issue occurred in extracting result metadata.
    */
-  protected void executeHelper() throws SQLException {
+  protected void executeHelper(int[] updateCounts) throws SQLException {
```

Comment:
This method has a quite generic name and it's unclear why it should get an updateCounts argument. The issue is probably pre-existing, but adding this argument here makes it worse.

One simple way that (imo) this would be more readable is if we update the this.updateCount at the callsite for batch statements and here we check if this.updateCount != null and use it directly, else we use this.statement.getUpdateCount()

Thoughts?

Reply:
We could also just rename executeHelper to something like processResults which better signifies what the intended purpose was.